### DISCUSSION TITLE
ECMWF historical forecast data

### DISCUSSION THREAD

**Discussion by adriankreiner** [microsoft/PlanetaryComputer#286] [Q&A] [answered]
Dear all, 
Thank you for your time helping me with my question! I am wondering if I can also access historical ECMWF weather forecast data. I found that it is possible to access the weather forecast under the following link: 
https://planetarycomputer.microsoft.com/dataset/ecmwf-forecast
Is there also a possibility to access past forecast data? 
Thanks a lot in advance! 
Best, 
Adrian

**Comment by TomAugspurger** [comment]
https://planetarycomputer.microsoft.com/api/stac/v1/search?collections=ecmwf-forecast&sortby=+datetime shows a search order by datetime ascending. We currently have data from 2022-01-21 onwards.

**Reply by adriankreiner** [reply]
Thank you very much! I appreciate your answer and it was really helpful! I attempted to extract the forecast (type forecast) for temperature (2t) and precipitation (tp) for a specific location using the R language, but I was not successful. Are there any example scripts using the MPC that demonstrate how to do this? I have already reviewed the linked example notebooks (https://github.com/ecmwf/notebook-examples/tree/master/opencharts) and how to  read from the STAC API with R (reading-stac-r.ipynb). Thanks a lot in advance!

**Comment by TomAugspurger** [comment]
The reading-stac-r document is the one I would have linked you too. I'm not sure beyond that, sorry.
________________________________

**Reply by adriankreiner** [reply]
Thanks a lot for the support! I have a follow up question. Is it possible to use Climate Data Operators (CDO) tool within the planetary computer?

**Comment by TomAugspurger** [comment]
I haven't used it before so I'm not sure. It isn't pre-installed in the environment, but you might be able to install it.
________________________________

**Reply by adriankreiner** [reply, marked as answer]
Thanks a lot! I will try to do this.
